File: app/src/main/java/com/example/games/RecyclerViewAdapter.java
package com.example.games;

import android.app.Dialog;
import android.content.Context;
import android.media.Image;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {

    Context mContext ;
    List<Games> mData;
    Dialog myDialog;

    public RecyclerViewAdapter(Context mContext, List<Games> mData) {
        this.mContext = mContext;
        this.mData = mData;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View v ;
        v = LayoutInflater.from(mContext).inflate(R.layout.list_item,parent,false);
        final MyViewHolder vHolder = new MyViewHolder(v);

        myDialog = new Dialog(mContext);
        myDialog.setContentView(R.layout.activity_games);

        vHolder.list_item.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TextView activity_heading = (TextView) myDialog.findViewById(R.id.heading_id);
                TextView activity_metacritic = (TextView) myDialog.findViewById(R.id.metacritic_id);
                TextView activity_genre = (TextView) myDialog.findViewById(R.id.genre_id);

                activity_heading.setText(mData.get(vHolder.getAdapterPosition()).getHeading());
                activity_metacritic.setText(mData.get(vHolder.getAdapterPosition()).getMetacritic());
                activity_genre.setText(mData.get(vHolder.getAdapterPosition()).getGenre());

                myDialog.show();
            }
        });

        return vHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {

        holder.img.setImageResource(mData.get(position).getTitleImage());
        holder.tv_heading.setText(mData.get(position).getHeading());
        holder.tv_metacritic.setText(mData.get(position).getMetacritic());
        holder.tv_genre.setText(mData.get(position).getGenre());

    }

    @Override
    public int getItemCount() {
        return mData.size();
    }


    public static class MyViewHolder extends RecyclerView.ViewHolder{

        private ConstraintLayout list_item;
        private ImageView img;
        private TextView tv_heading;
        private TextView tv_metacritic;
        private TextView tv_genre;


        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            list_item = (ConstraintLayout) itemView.findViewById(R.id.list_item_id);
            img = (ImageView) itemView.findViewById(R.id.title_image);
            tv_heading = (TextView) itemView.findViewById(R.id.tvheading);
            tv_metacritic = (TextView) itemView.findViewById(R.id.metacritic);
            tv_genre = itemView.findViewById(R.id.genre);
        }
    }

}

File: app/src/main/java/com/example/games/RecyclerViewAdapter.kt
package com.example.games

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import com.example.games.RecyclerViewAdapter.MyViewHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.AppCompatImageView
import androidx.swiperefreshlayout.widget.CircularProgressDrawable
import com.bumptech.glide.Glide

class RecyclerViewAdapter( var gamesList: List<Games>, private val adapterContext: Context?) :
    RecyclerView.Adapter<MyViewHolder>() {

    private lateinit var  mListener : onItemClickListener

    interface onItemClickListener{

        fun onItemClick(position: Int)

    }

    fun setOnItemClickListener(listener : onItemClickListener){

        mListener = listener

    }

    fun setFilteredList(gamesList: List<Games>){
        this.gamesList = gamesList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.list_item, parent, false)
        return MyViewHolder(v,mListener)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val currentItem = gamesList[position]
        val circularProgressDrawable = CircularProgressDrawable(adapterContext!!)

        circularProgressDrawable.strokeWidth = 5f
        circularProgressDrawable.centerRadius = 30f
        circularProgressDrawable.start()

        Glide.with(holder.img.context)
            .load(currentItem.titleImage)
            .placeholder(circularProgressDrawable)
            .into(holder.img)

        holder.tv_heading.text = currentItem.heading
        holder.tv_metacritic.text = currentItem.metacritic
        holder.tv_genre.text = currentItem.genre
    }

    override fun getItemCount(): Int {
        return gamesList.size
    }

    class MyViewHolder(itemView: View, listener: onItemClickListener) : RecyclerView.ViewHolder(itemView) {

        var img : ImageView = itemView.findViewById(R.id.title_image)
        val tv_heading : TextView = itemView.findViewById(R.id.tvheading)
        val tv_metacritic : TextView = itemView.findViewById(R.id.metacritic)
        val tv_genre : TextView = itemView.findViewById(R.id.genre)

        init {

            itemView.setOnClickListener {

                listener.onItemClick(adapterPosition)

            }

        }
    }
}